state/errors/common.go

```diff
@@ -165,49 +169,13 @@ func IsParentDeviceHasChildrenError(err interface{}) bool {
 	return ok
 }
 
-// errIncompatibleSeries is a standard error to indicate that the series
-// requested is not compatible with the charm of the application.
-type errIncompatibleSeries struct {
-	seriesList []string
-	series     string
-	charmName  string
-}
-
 func NewErrIncompatibleSeries(seriesList []string, series, charmName string) error {
-	return &errIncompatibleSeries{
-		seriesList: seriesList,
-		series:     series,
-		charmName:  charmName,
-	}
-}
-
-func (e *errIncompatibleSeries) Error() string {
-	return fmt.Sprintf("series %q not supported by charm %q, supported series are: %s",
-		e.series, e.charmName, strings.Join(e.seriesList, ", "))
-}
-
-// IsIncompatibleSeriesError returns if the given error or its cause is
-// errIncompatibleSeries.
-func IsIncompatibleSeriesError(err interface{}) bool {
-	if err == nil {
-		return false
-	}
-	// In case of a wrapped error, check the cause first.
-	value := err
-	cause := errors.Cause(err.(error))
-	if cause != nil {
-		value = cause
-	}
-	_, ok := value.(*errIncompatibleSeries)
-	return ok
-}
-
-var ErrUpgradeInProgress = errors.New("upgrade in progress")
-
-// IsUpgradeInProgressError returns true if the error is caused by an
-// in-progress upgrade.
-func IsUpgradeInProgressError(err error) bool {
-	return errors.Cause(err) == ErrUpgradeInProgress
+	return fmt.Errorf("series %q not supported by charm %q, supported series are: %s%w",
+		series,
+		charmName,
+		strings.Join(seriesList, ", "),
+		errors.Hide(IncompatibleSeriesError),
+	)
```

Comment:
The `%s%w` with no space/separator in between looks weird and then you need to see that Hide() is used and can then eventually figure out what's going on.
I with we could use Wrap() to do:

```
return errors.Wrap(IncompatibleSeriesError, fmt.Errorf("series %q not supported by charm %q, supported series are: %s", series, charmName, strings.Join(seriesList, ", "))
```

which internally could do the Hide() thing but it sure reads much better